src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -34,6 +36,51 @@
 #include <bsoncxx/oid.hpp>
 #include <bsoncxx/types/bson_value/view.hpp>
 
+namespace {
+
+template <class RealType = int64_t>
+class zipfian_distribution {
+public:
+    explicit zipfian_distribution(double alpha, RealType n, double c)
+        : _alpha{alpha}, _n{n}, _c{c} {}
+
+    template <class URNG>
+    RealType operator()(URNG& urng) {
+        return generate(urng);
+    }
+
+private:
+    // Shape parameter for the distribution.
+    double _alpha;
+    // Normalization constant for the distribution.
+    double _c;
+    // Number of distinct elements in the distribution.
+    RealType _n;
+
+    // This implementation is based on https://cse.usf.edu/~kchriste/tools/genzipf.c, 
+    // it uses the inverse transform sampling method for generating random numbers.
+    // This method is not the most accurate and efficient for heavy tailed distributions,
+    // but is sufficient for our current purposes.
+    template<class URNG>
+    RealType generate(URNG& urng) {
+        double sum = 0;
+        double randomNumber = 0;
+        
+        while (randomNumber <= 0) {
+            randomNumber = (double)urng() / (double)urng.max();
```

Comment:
This doesn't seem correct.  Shouldn't you pass in a URNG that handles this for you?

Reply:
suggestion: change to something like the following:

```
        boost::random::uniform_01<> _uniform01{};
        double randomNumber = _uniform01(urng);
```